### PR TITLE
Add training provider and accredited provider filters to the provider show page

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

This PR does 3 things

1. adds a training provider filter
2. adds an accredited provider filter
3. refactors out most of the logic into a helper 

The refactor was necessary as we don't want to expose the training or accrediting provider filter on the candidates application page (it would be 2000 checkboxes or so long 😱)

## Changes proposed in this pull request

- Adds a training provider filter to the providers applications page which returns the provider who runs the course
- Adds a training provider filter to the providers applications page which returns a list of courses for the accrediting provider
- Decouples the two filter interfaces with a helper

## Guidance to review

Per commit is going to be easiest to follow the thought process 

I think one of the queries maybe slightly off. I think the accredited provider is returning a list or providers who accredit one of the training providers courses, rather than ones with applications.

My brain is mush though.

## Link to Trello card

https://trello.com/c/LGclXXrY/3827-add-courses-run-by-filter-to-provider-applications-page-on-support-console
https://trello.com/c/oDVIHY4j/3828-add-courses-ratified-by-filter-to-provider-applications-page-on-support-console

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/helpers/application_filter_helper.rb [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    }
+  end
+
+  def accredited_provider_filter
+    courses = Course
+              .joins(:provider)
+              .includes([:accredited_provider])
+              .where(provider: { id: applied_filters['provider_id'] })
+              .or(Course.where(accredited_provider_id: applied_filters['provider_id']))
+              .distinct
```
Comment (davidgisbey) [inline app/helpers/application_filter_helper.rb:159]: I think this query is actually wrong as it should really only be providing accrediting providers where a candidate has actually applied to their course. Ngl my brain is absolutely melted though.

Help me @stevehook .... you're my only hope

File: app/helpers/application_filter_helper.rb [hunk 2]
```diff
@@ -0,0 +1,248 @@
+module ApplicationFilterHelper
```
Comment (TobyRet) [inline app/helpers/application_filter_helper.rb:1]: Not a blocker for merging but wonder if this should be in `models` as well ... I guess technically this is more like a query service? Either way `app/helpers' may not be the best home for this ...

File: app/helpers/application_filter_helper.rb (hunk 1 above)
Comment (TobyRet) [reply app/helpers/application_filter_helper.rb:159]: Sorry I know we discussed this @davidgisbey but now that I've taken another a look it does seem like we would only want to return accredited providers for this filter right?

**Comment by TobyRet** [comment]
Had a play around with locally and it seem to work well. Only minor suggestion is whether we want to hide the training provider filter if the provider in question is the training provider and hide the accredited providers filter if the provider is the accredited provider 🤷‍♂️

File: app/helpers/application_filter_helper.rb (hunk 1 above)
Comment (stevehook) [reply app/helpers/application_filter_helper.rb:159]: Still trying to understand this properly but it makes sense to limit this set of options to accredited providers that have received applications. Doesn't the same thing apply to `training_provider_filter` above?

**Comment by davidgisbey** [comment]
I'll pick this up after we're done with the reporting work

**Pull request closed without merging**